fileUtils.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('fs');
+const lockMap = new Map();
+
+const writeFile = (path, src) => {
+    return new Promise(async (resolve, reject) => {
+        try {
+            if (lockMap.get(key) && !lockMap.get(key).locked) {
+                const fileHandle = await fs.promises.open(`${process.cwd()}/chunkSrc/${path}`, 'w+');
+                if (fileHandle) {
+                    await fs.promises.writeFile(fileHandle, src);
```

Comment:
I believe the file isn't being closed? So it's a handle leak.

Reply:
a204dec